### PR TITLE
feat(lsp): add neovim-lsp

### PR DESCRIPTION
**Pull request by wsdjeg**

### CODE REVIEW

**Comment by vuvgaG** [comment]
Is this general improvements to the LSP or are you adding just lua language?

File: autoload/SpaceVim/layers/lsp.vim [hunk 1]
```diff
@@ -6,19 +6,31 @@
...
 " This layer provides language client support for SpaceVim.
 
+let s:NVIM_VERSION = SpaceVim#api#import('neovim#version')
+let s:lsp_client = ''
+
 function! SpaceVim#layers#lsp#plugins() abort
   let plugins = []
 
-  if SpaceVim#layers#isLoaded('autocomplete') && get(g:, 'spacevim_autocomplete_method') ==# 'coc'
+  if (has('nvim-0.5.0') && s:NVIM_VERSION.is_release_version()) || has('nvim-0.6.0')
```
Comment (vuvgaG) [inline autoload/SpaceVim/layers/lsp.vim:82]: This check won't catch bug fixes etc???

**Comment by xusd320** [comment]
when will this feature be ready?

File: autoload/SpaceVim/layers/lsp.vim (hunk 1 above)
Comment (wsdjeg) [reply autoload/SpaceVim/layers/lsp.vim:82]: which bug？

**Pull request merged**